custom/userAuth.js

```diff
@@ -50,7 +52,34 @@ getAuth().then(({email, key}) => {
 
   // seralize/deseralization methods for extracting user information from the
   // session cookie and adding it to the req.passport object
-  passport.serializeUser((user, done) => done(null, user))
+
+  // serialize is called when the user first authenticates
+  // Use this callback as an opportunity to check their email against the chive permissions list
+  // and attach an "authorized" flag to their PassportJS object
+  passport.serializeUser(async (user, done) => {
+    user.authorized = false;
+
+    // magic Google Apps Script; takes a "user" parameter in a GET request and returns a JSON object with an "authorized" key
+    const authUrl = "https://script.google.com/macros/s/AKfycby3zj7uw2ZHi6JUcCt8aaRoE3Z8lUo3uKojxGXXIgeCoJ9h78CvXGJasUl--t23JfxB/exec"
+
+    try {
+      // TODO: Technically there could be more than one email attached to a profile - should we send all of them?
+      var authResp = await axios.get(authUrl, {params: {user: user.emails[0].value}})
+
+      if (authResp.data.authorized) {
+        user.authorized = true;
+      }
+    } catch (e) {
+      console.log(e)
+    }
+
+    // save the user object and finish serializing it
+    done(null, user)
+  })
+
+  // deserialize is called anytime the user loads a new page; no need to re-check authorization
+  // However, if user is added to permissions list *after* they first log in, they will need to re-log-in
```

Comment:
Hm this makes me wonder about the case where a user is *removed* from the list but still holds a session from when they were on the list. Not a reason to block merge, since this PR is still an improvement over the existing situation, but maybe add a note/TODO in the code that this would be nice to add.

Reply:
I thought about this, and I was kinda assuming that if that ever happens and we care strongly about them losing access we can invalidate everyone's session. I'll add a TODO, as there's not a simple way to fix it (I guess you'd need to store a time-since-last-login)

Reply:
Added TODO.